tests/trx_generator/trx_generator_tests.cpp

```diff
@@ -0,0 +1,192 @@
+   constexpr uint64_t minimum_runtime_us = expected_runtime_us - allowable_runtime_deviation_us;
+   constexpr uint64_t maximum_runtime_us = expected_runtime_us + allowable_runtime_deviation_us;
+
+   std::shared_ptr<mock_trx_generator> generator = std::make_shared<mock_trx_generator>(expected_trxs);
+   std::shared_ptr<simple_tps_monitor> monitor = std::make_shared<simple_tps_monitor>(expected_trxs);
+
+
+   trx_tps_tester<mock_trx_generator, simple_tps_monitor> t1(generator, monitor, test_duration_s, test_tps);
+
+   fc::time_point start = fc::time_point::now();
+   t1.run();
+   fc::time_point end = fc::time_point::now();
+   fc::microseconds runtime_us = end.time_since_epoch() - start.time_since_epoch() ;
+
+   BOOST_REQUIRE_EQUAL(generator->_calls.size(), expected_trxs);
+   BOOST_REQUIRE_GT(runtime_us.count(), minimum_runtime_us);
+
+   if (runtime_us.count() > maximum_runtime_us) {
+      ilog("couldn't sustain transaction rate.  ran ${rt}us vs expected max ${mx}us",
+           ("rt", runtime_us.count())("mx", maximum_runtime_us ) );
+      BOOST_REQUIRE_LT(monitor->_calls.back().last_sleep, 0);
+   }
+
+}
+
+BOOST_AUTO_TEST_CASE(tps_short_run_med_tps_med_delay)
+{
+   constexpr uint32_t test_duration_s = 5;
+   constexpr uint32_t test_tps = 10000;
+   constexpr uint32_t trx_delay_us = 10;
+   constexpr uint32_t expected_trxs = test_duration_s * test_tps;
+   constexpr uint64_t expected_runtime_us = test_duration_s * 1000000;
+   constexpr uint64_t allowable_runtime_deviation_per = 20;
+   constexpr uint64_t allowable_runtime_deviation_us = expected_runtime_us / allowable_runtime_deviation_per;
+   constexpr uint64_t minimum_runtime_us = expected_runtime_us - allowable_runtime_deviation_us;
+   constexpr uint64_t maximum_runtime_us = expected_runtime_us + allowable_runtime_deviation_us;
+
+   std::shared_ptr<mock_trx_generator> generator = std::make_shared<mock_trx_generator>(expected_trxs, trx_delay_us);
+   std::shared_ptr<simple_tps_monitor> monitor = std::make_shared<simple_tps_monitor>(expected_trxs);
+
+
+   trx_tps_tester<mock_trx_generator, simple_tps_monitor> t1(generator, monitor, test_duration_s, test_tps);
+
+   fc::time_point start = fc::time_point::now();
+   t1.run();
+   fc::time_point end = fc::time_point::now();
+   fc::microseconds runtime_us = end.time_since_epoch() - start.time_since_epoch() ;
+
+   BOOST_REQUIRE_EQUAL(generator->_calls.size(), expected_trxs);
+   BOOST_REQUIRE_GT(runtime_us.count(), minimum_runtime_us);
+
+   if (runtime_us.count() > maximum_runtime_us) {
+      ilog("couldn't sustain transaction rate.  ran ${rt}us vs expected max ${mx}us",
+           ("rt", runtime_us.count())("mx", maximum_runtime_us ) );
+      BOOST_REQUIRE_LT(monitor->_calls.back().last_sleep, 0);
+   }
+}
+
+BOOST_AUTO_TEST_CASE(tps_med_run_med_tps_med_delay)
+{
+   constexpr uint32_t test_duration_s = 30;
+   constexpr uint32_t test_tps = 10000;
+   constexpr uint32_t trx_delay_us = 10;
```

Comment:
Would it be valuable to add a test where the `trx_delay_us` is higher?

Reply:
Yep, do you have a suggested delay?

Reply:
I don't honestly.  Just wasn't sure if we should test multiple values just to have some variability in testing.